pkg/vadmin/re_ip_vc.go

```diff
@@ -18,15 +18,74 @@ package vadmin
 import (
 	"context"
 	"fmt"
+	"strings"
 
+	vops "github.com/vertica/vcluster/vclusterops"
+	"github.com/vertica/vcluster/vclusterops/vstruct"
+	vapi "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/net"
 	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/reip"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 // ReIP will update the catalog on disk with new IPs for all of the nodes given.
 func (v *VClusterOps) ReIP(ctx context.Context, opts ...reip.Option) (ctrl.Result, error) {
 	v.Log.Info("Starting vcluster ReIP")
+
+	// get re-ip options
 	s := reip.Parms{}
 	s.Make(opts...)
-	return ctrl.Result{}, fmt.Errorf("not implemented")
+
+	// call vcluster-ops library to re-ip
+	vopts, err := v.genReIPOptions(&s)
+	if err != nil {
+		v.Log.Error(err, "failed to set up re-ip options")
+		return ctrl.Result{}, err
+	}
+
+	err = v.VReIP(&vopts)
+	if err != nil {
+		v.Log.Error(err, "failed to run re-ip")
+		return ctrl.Result{}, err
+	}
+
+	v.Log.Info("Successfully complete re-ip")
+	return ctrl.Result{}, nil
+}
+
+func (v *VClusterOps) genReIPOptions(s *reip.Parms) (vops.VReIPOptions, error) {
+	opts := vops.VReIPFactory()
+
+	// hosts
+	for _, host := range s.Hosts {
+		opts.Hosts = append(opts.Hosts, host.IP)
+	}
+	v.Log.Info("Setup re-ip options", "hosts", strings.Join(opts.Hosts, ","))
+	if len(opts.Hosts) == 0 {
+		return vops.VReIPOptions{}, fmt.Errorf("hosts should not be empty")
+	}
+
+	// ipv6
+	opts.Ipv6 = vstruct.MakeNullableBool(net.IsIPv6(opts.Hosts[0]))
+
+	// catalog prefix
+	*opts.CatalogPrefix = v.VDB.Spec.Local.GetCatalogPath()
+
+	// database name
+	opts.Name = &v.VDB.Spec.DBName
+
+	// re-ip list
+	for _, h := range s.Hosts {
+		var reIPInfo vops.ReIPInfo
+		reIPInfo.NodeName = h.VNode
+		reIPInfo.TargetAddress = h.IP
+		reIPInfo.NodeAddress = h.IP
```

Comment:
Why do we need to fill in this field? How is it different then the TargetAddress?

Reply:
We don't need it. This is just to avoid an incorrect duplicate IP checking on the vcluster side.
I will remove this line.